src/corelibs/U2View/src/ov_sequence/PanViewRows.cpp

```diff
@@ -86,22 +94,12 @@ void PVRowsManager::addAnnotation(Annotation* a) {
     const SharedAnnotationData& data = a->getData();
     const QVector<U2Region> location = data->getRegions();
 
-    QString name;
-    bool isRestrictionSite = data->type == U2FeatureTypes::RestrictionSite;
-    if (isRestrictionSite) {
-        name = QObject::tr("Restriction Site");
-    } else {
-        name = data->name;
-    }
-
-    if (rowByName.contains(name)) {
-        foreach (PVRowData* row, rowByName[name]) {
+    QString name = data->type == U2FeatureTypes::RestrictionSite ? PVRowData::RESTRICTION_SITE_NAME : data->name;
```

Comment:
This code is duplicated: please extract into getAnnotationRowName(Annotation*)

Reply:
Now its not duplicated